src/Sylius/Bundle/ThemeBundle/Asset/Installer/AssetsInstaller.php

```diff
@@ -0,0 +1,251 @@
+     * @param string $originDir
+     * @param string $targetDir
+     * @param integer $symlinkMask
+     *
+     * @return integer
+     */
+    protected function installThemedBundleAssets(ThemeInterface $theme, $originDir, $targetDir, $symlinkMask)
+    {
+        $effectiveSymlinkMask = $symlinkMask;
+
+        $finder = new Finder();
+        $finder->sortByName()->ignoreDotFiles(false)->in($originDir);
+
+        /** @var SplFileInfo[] $finder */
+        foreach ($finder as $originFile) {
+            $targetFile = $targetDir . '/' . $originFile->getRelativePathname();
+            $targetFile = $this->pathResolver->resolve($targetFile, $theme);
+
+            if (file_exists($targetFile)) {
+                continue;
+            }
+
+            $this->filesystem->mkdir(dirname($targetFile));
+
+            $effectiveSymlinkMask = min(
+                $effectiveSymlinkMask,
+                $this->installAsset($originFile->getPathname(), $targetFile, $symlinkMask)
+            );
+        }
+
+        return $effectiveSymlinkMask;
+    }
+
+    /**
+     * @param string $originDir
+     * @param string $targetDir
+     * @param integer $symlinkMask
+     *
+     * @return integer
+     */
+    protected function installVanillaBundleAssets($originDir, $targetDir, $symlinkMask)
+    {
+        return $this->installAsset($originDir, $targetDir, $symlinkMask);
+    }
+
+    /**
+     * @param string $origin
+     * @param string $target
+     * @param integer $symlinkMask
+     *
+     * @return integer
+     */
+    protected function installAsset($origin, $target, $symlinkMask)
+    {
+        if (AssetsInstallerInterface::RELATIVE_SYMLINK === $symlinkMask) {
+            try {
+                $targetDirname = realpath(is_dir($target) ? $target : dirname($target));
+                $relativeOrigin = rtrim($this->filesystem->makePathRelative($origin, $targetDirname), '/');
+
+                $this->doInstallAsset($relativeOrigin, $target, true);
+
+                return AssetsInstallerInterface::RELATIVE_SYMLINK;
+            } catch (IOException $exception) {
+                // Do nothing, trying to create non-relative symlinks later.
+            }
+        }
+
+        if (AssetsInstallerInterface::HARD_COPY !== $symlinkMask) {
+            try {
+                $this->doInstallAsset($origin, $target, true);
+
+                return AssetsInstallerInterface::SYMLINK;
+            } catch (IOException $exception) {
+                // Do nothing, hard copy later.
+            }
+        }
+
+        $this->doInstallAsset($origin, $target, false);
+
+        return AssetsInstallerInterface::HARD_COPY;
+    }
+
+    /**
+     * @param string $origin
+     * @param string $target
+     * @param boolean $symlink
+     *
+     * @throws IOException When failed to make symbolic link, if requested.
+     */
+    protected function doInstallAsset($origin, $target, $symlink)
+    {
+        if ($symlink) {
```

Comment:
Consider reversing condition, to eliminate nested if and else statements. 
```php 
if ($symlink) {
    $this->filesystem->symlink($origin, $target);
    
    if (!file_exists($target)) {
        throw new IOException('Symbolic link is broken');
    }
    
    return;
}

if (is_dir($origin)) {
    $this->filesystem->mkdir($target, 0777);
    $this->filesystem->mirror($origin, $target, Finder::create()->ignoreDotFiles(false)->in($origin));
    
    return;
}

$this->filesystem->copy($origin, $target);
```
or even move if($symlink) body to private method, to avoid nested if.